packages/kilt-did/src/types.ts

```diff
@@ -0,0 +1,104 @@
+import type { IIdentity, SubmittableExtrinsic } from '@kiltprotocol/types'
+import { u64 } from '@polkadot/types'
+import { AnyNumber } from '@polkadot/types/types'
+import { KeyId } from './types.chain'
+
+export type KeypairType = string
+
+export interface IPublicKey {
+  publicKey: Uint8Array
+  type: KeypairType
+  // do we need the identifier?
+}
+
+export type IKeyPair = IPublicKey
+
+export type ISigningKey = IPublicKey
+export type IEncryptionKey = IPublicKey
```

Comment:
If we use `IEncryptionKey` and `IEncryptionKeyPair`, then I think it would make more sense to rename `IEncryptionKey` to `IEncryptionPublicKey`. For `ISigningKey`, the right terminology would be `IVerificationKey` for the public key, and `ISigningKey` for the private one.